src/main/drivers/barometer/barometer_dps310.c

```diff
@@ -173,19 +173,21 @@ static bool deviceConfigure(const extDevice_t *dev)
     // 1. Read the pressure calibration coefficients (c00, c10, c20, c30, c01, c11, and c21) from the Calibration Coefficient register.
     //   Note: The coefficients read from the coefficient register are 2's complement numbers.
     // Do the read of the coefficients in multiple parts, as the chip will return a read failure when trying to read all at once over I2C.
-#ifndef USE_BARO_DPS310_GOERTEK
-#define COEFFICIENT_LENGTH 18
-#else
-#define COEFFICIENT_LENGTH 22
-#endif
+    uint8_t COEFFICIENT_LENGTH, READ_LENGTH;
 
-#define READ_LENGTH (COEFFICIENT_LENGTH / 2)
+    if (chipId[0] == SPL07_003_CHIP_ID) {
+        COEFFICIENT_LENGTH = 22;
+    } else {
+        COEFFICIENT_LENGTH = 18;
+    }
+
+    READ_LENGTH = COEFFICIENT_LENGTH / 2;
 
     uint8_t coef[COEFFICIENT_LENGTH];
     if (!busReadBuf(dev, DPS310_REG_COEF, coef, READ_LENGTH)) {
         return false;
     }
-     if (!busReadBuf(dev, DPS310_REG_COEF + READ_LENGTH, coef + READ_LENGTH, COEFFICIENT_LENGTH - READ_LENGTH)) {
+    if (!busReadBuf(dev, DPS310_REG_COEF + READ_LENGTH, coef + READ_LENGTH, COEFFICIENT_LENGTH - READ_LENGTH)) {
         return false;
     }
```

Comment:
```
for (unsigned i = 0; i < sizeof(coef); ) {
   int chunk = MIN(9, sizeof(coef) - i);
   if (!busReadBuf(dev, DPS310_REG_COEF + i,  coef + i, chunk) {
      return false;
   }
   i += chunk;
}

Reply:
Instead of 9 we need READ_LENGTH (COEFFICIENT_LENGTH / 2) ???

Reply:
Why? It seems someone decided reading all 18 bytes is too much. So it was split into 2 chunks. 
IMO it is pretty arbitrary and reading all bytes shall work, but it is safer to keep tested readout mechanism.